community/detectors/jira_cve_2022_0540/src/main/java/com/google/tsunami/plugins/detectors/jira/Cve20220540VulnDetector.java

```diff
@@ -0,0 +1,177 @@
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      // Assume the service uses HTTP protocol when the scanner cannot identify the actual service.
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(INSIGHT_CHECK_VUL_PATH);
+    return targetUrlBuilder.toString();
+  }
+
+  private static String buildTargetWBSGanttUrl(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      // Assume the service uses HTTP protocol when the scanner cannot identify the actual service.
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(WBSGANTT_CHECK_VUL_PATH);
+    return targetUrlBuilder.toString();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve20220540VulnDetector starts detecting.");
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve20220540VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String insightUrl = buildTargetInsightUrl(networkService);
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(insightUrl).withEmptyHeaders().build(), networkService);
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(INSIGHT_BODY)) {
+        return true;
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+
+    String WBSGanttUrl = buildTargetWBSGanttUrl(networkService);
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(WBSGanttUrl).withEmptyHeaders().build(), networkService);
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(WBSGANTT_DOBY)) {
+        return true;
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
```

Comment:
Since the detection is dependent on specific plugin installed, could you also pass in the plugin details into the description?

You can refactor isServiceVulnerable to also take in a targetUrl (either insight or WBSGantt) to dedupe the detection logic.